packages/vue-components/src/SiteNavButton.vue

```diff
@@ -19,21 +16,16 @@ import $ from './utils/NodeList';
 export default {
   data() {
     return {
-      hasIdentifier: false,
-      hasSiteNav: false,
+      portalName: undefined,
       show: false,
-      src: '',
     };
   },
   computed: {
     showSiteNav() {
-      return this.show && this.hasSiteNav;
+      return this.show && this.portalName;
     },
   },
   methods: {
-    getSiteNavContent() {
-      return document.getElementsByClassName('site-nav-root')[0];
-    },
```

Comment:
Similar to page nav (Should pull the first site nav if the identifier is not present).

Reply:
also now using the portal implementation (more important here as `<site-nav>` can contain interactive vue components)